functions/nfts/asset/[[index]].ts

```diff
@@ -9,11 +9,20 @@ export const onRequest: PagesFunction = async ({ params, request, next }) => {
   const assetPromise = getAsset(collectionAddress, tokenId, request.url)
   const resPromise = next()
   try {
-    const [data, res] = await Promise.all([assetPromise, resPromise])
-    if (!data) {
-      return resPromise
+    const cache = caches.default
+    const response = await cache.match(request.url)
+    if (response) {
+      const data = JSON.parse(await response.text())
+      return new HTMLRewriter().on('head', new MetaTagInjector(data)).transform(await resPromise)
+    } else {
+      const [data, res] = await Promise.all([assetPromise, resPromise])
+      if (!data) {
+        return resPromise
+      }
+      const response = new Response(JSON.stringify(data))
+      await cache.put(request.url, response)
+      return new HTMLRewriter().on('head', new MetaTagInjector(data)).transform(res)
```

Comment:
You should cache the response _with_ the meta tags already injected to avoid having to redo the work.